capstone/src/main/webapp/sheetsScript.js

```diff
@@ -62,29 +62,302 @@ async function gatherSheetsInput() {
     }
   }
 
-  const text = await getTextFromSheet(id);
-  handleInput(text, langElement.value, requestedAttributes, tokenizer);
+  // Get the name and range of sheets to be analyzed
+  const response = await getSheetNamesAndRange(id);
+  const sheetNames = await response.sheetNames;
+  const range = await response.range;
```

Comment:
nit: I think you may be able to omit these `await`s because you already `await` for the response in the line above.

Reply:
Sounds good - I'll remove them.